### ISSUE TITLE
Indicate whether metric is counter or gauge in Prometheus exposition format

### ISSUE THREAD

**Issue by l2dy** [iopsystems/rezolus#8]
Counter metrics should have `counter` TYPE and have `_total` suffix following Prometheus convention.

https://github.com/iopsystems/rezolus/blob/fd7f7e2f39db3f000c5d62aaeda0f696596f434d/src/exposition/mod.rs#L48

### Expected behavior

1. TYPE should be `counter` for counters like `cpu_usage_idle`, but `cpu_frequency` should still be `gauge` TYPE.
2. Add a fixed suffix of `_total` to all counter metrics in `prometheus(&self)`, for example from `cpu_usage_idle` to `cpu_usage_idle_total`.

### Actual behavior

Both metric are exposed as `gauge` TYPE, and no suffix is appended.

### Steps to reproduce the behavior

1. Run `rezolus` with example config and scrape `/metrics`.

**Comment by brayniac** [comment]
Thanks for opening this. We are in the middle of a ground-up rewrite of Rezolus and will have the correct `TYPE` annotations on counters in the rewrite.

I'm less sure the `_total` suffix convention is universally correct. For example, the metric you listed should have the `_nanoseconds_total` suffix per the conventions. `idle` would be better represented as a label. I'm not totally convinced that there's value in adding the suffixes suggested in the convention if we have the right type annotations.

**Comment by l2dy** [comment]
> Thanks for opening this. We are in the middle of a ground-up rewrite of Rezolus and will have the correct `TYPE` annotations on counters in the rewrite.

You are welcome!

> For example, the metric you listed should have the `_nanoseconds_total` suffix per the conventions. `idle` would be better represented as a label.

Yes, that would be more conformant to https://prometheus.io/docs/practices/naming/ rules.

> I'm not totally convinced that there's value in adding the suffixes suggested in the convention if we have the right type annotations.

Prometheus supports metadata query since [v2.15+](https://github.com/prometheus/prometheus/blob/v2.15.0/CHANGELOG.md), but the ecosystem is yet to catch up, for example VictoriaMetrics has not implemented the endpoint yet: `https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2974` (quoted to avoid mention references on GitHub).

Adding the suffix helps Grafana, which checks metric suffix if metadata is not available:

https://grafana.com/blog/2020/06/15/how-we-made-working-with-prometheus-easier-with-metric-metadata-in-grafanas-explore-view/
https://github.com/grafana/grafana/blob/9041e7dfa6c316c7db71f6fba1845b94ad07fb50/public/app/plugins/datasource/prometheus/query_hints.ts#L35-L79

**Comment by brayniac** [comment]
@l2dy - thanks for the detail on that. Will consider adding at least the `_total` suffix to counters in the rewrite. I haven't run into problems personally, but I've mainly been using Prometheus + Grafana so didn't realize that the rest of the ecosystem was lagging in terms of the metadata support.

Thanks again for opening this, I'll update once the rewrite is merged into main. Anticipating that we'll be ready for that in the next couple of weeks.

**Comment by brayniac** [comment]
I think this has all been addressed in the latest releases.
